tests/test_analysis.py

```diff
@@ -0,0 +1,126 @@
+             {u'dates': {u'buckets':
+                             [{u'key': 1388534400000, u'doc_count': 0, u'key_as_string': u'2014-01-01'},
+                              {u'key': 1391212800000, u'doc_count': 0, u'key_as_string': u'2014-02-01'},
+                              {u'key': 1393632000000, u'doc_count': 0, u'key_as_string': u'2014-03-01'},
+                              {u'key': 1396310400000, u'doc_count': 0, u'key_as_string': u'2014-04-01'},
+                              {u'key': 1398902400000, u'doc_count': 0, u'key_as_string': u'2014-05-01'},
+                              {u'key': 1401580800000, u'doc_count': 0, u'key_as_string': u'2014-06-01'},
+                              {u'key': 1404172800000, u'doc_count': 0, u'key_as_string': u'2014-07-01'},
+                              {u'key': 1406851200000, u'doc_count': 0, u'key_as_string': u'2014-08-01'},
+                              {u'key': 1409529600000, u'doc_count': 0, u'key_as_string': u'2014-09-01'},
+                              {u'key': 1412121600000, u'doc_count': 0, u'key_as_string': u'2014-10-01'},
+                              {u'key': 1414800000000, u'doc_count': 0, u'key_as_string': u'2014-11-01'},
+                              {u'key': 1417392000000, u'doc_count': 1685, u'key_as_string': u'2014-12-01'},
+                              {u'key': 1420070400000, u'doc_count': 13871, u'key_as_string': u'2015-01-01'},
+                              {u'key': 1422748800000, u'doc_count': 10175, u'key_as_string': u'2015-02-01'},
+                              {u'key': 1425168000000, u'doc_count': 14745, u'key_as_string': u'2015-03-01'},
+                              {u'key': 1427846400000, u'doc_count': 13881, u'key_as_string': u'2015-04-01'},
+                              {u'key': 1430438400000, u'doc_count': 16696, u'key_as_string': u'2015-05-01'},
+                              {u'key': 1433116800000, u'doc_count': 10784, u'key_as_string': u'2015-06-01'}]},
+              u'key': u'article', u'doc_count': 81837}]
+         }})
+
+    search_with_two_dim_aggs.execute = Mock(return_value=search_mock)
+    search_results = search_with_two_dim_aggs.execute()
+    two_dim_dataframe = agg_to_two_dim_dataframe(search_results.testing_two_dim_aggs)
+    print two_dim_dataframe
+    assert isinstance(two_dim_dataframe, pd.DataFrame)
+    assert two_dim_dataframe.shape[1] == 4
+    assert two_dim_dataframe.shape[0] == 18
+    for index,bucket in enumerate(search_results.testing_two_dim_aggs.buckets):
+        for index_2nd_level, raw_date in enumerate(two_dim_dataframe['dates.key']):
+            assert raw_date == search_results.testing_two_dim_aggs.buckets[index].dates.buckets[index_2nd_level]['dates.key']
+    for index_2nd_level, count_num in enumerate(two_dim_dataframe['dates.and']):
+        assert count_num == search_results.testing_two_dim_aggs.buckets[0].dates.buckets[index_2nd_level]['dates.and']
+
+#run tests
+test_basic_aggregation()
+test_two_dim_aggregation()
```

Comment:
The tests will actually be automatically discovered and run by pytest, so there is no need to run them manually here. Also, you should set PyCharm to automatically add newlines to the end of files.